### PR TITLE
Np 46/implement pdf export support

### PR DESCRIPTION
**Pull request by olhabn**
@kgrgreer @jlhughes 
Please review

### CODE REVIEW

File: src/foam/nanos/export/GoogleApiAuthService.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        LocalServerReceiver receiver = new LocalServerReceiver.Builder().setPort(credentialsConfig.getPort()).build();
+        return new AuthorizationCodeInstalledApp(flow, receiver).authorize("user");
+      `
+    }
+  ],
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function(cls) {
+        cls.extras.push(`private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();`);
```
Comment (kgrgreer) [inline src/foam/nanos/export/GoogleApiAuthService.js]: You could make this be a constant instead. Using constants:

File: src/foam/nanos/export/GoogleDriveService.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+        service.files().delete(fileId).execute();
+      `
+    }
+  ],
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function(cls) {
+        cls.extras.push(`private static final JsonFactory JSON_FACTORY = JacksonFactory.getDefaultInstance();
+        private static final List<String> SCOPES = Collections.singletonList(DriveScopes.DRIVE);`);
```
Comment (kgrgreer) [inline src/foam/nanos/export/GoogleDriveService.js]: Could be a constant.

File: src/foam/nanos/export/GoogleSheetsExportService.js [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+        }
+      `
+    }
+  ],
+  axioms: [
+    {
+      name: 'javaExtras',
+      buildJavaClass: function(cls) {
+        cls.extras.push(`
+        private static final List<String> SCOPES = Collections.singletonList(SheetsScopes.DRIVE_FILE);
```
Comment (kgrgreer) [inline src/foam/nanos/export/GoogleSheetsExportService.js]: again, use constants

File: src/foam/nanos/export/GoogleDriveService.js (hunk 2 above)
Comment (olhabn) [reply src/foam/nanos/export/GoogleDriveService.js]: fixed

File: src/foam/nanos/export/GoogleApiAuthService.js (hunk 1 above)
Comment (olhabn) [reply src/foam/nanos/export/GoogleApiAuthService.js]: fixed

File: src/foam/nanos/export/GoogleSheetsExportService.js (hunk 3 above)
Comment (olhabn) [reply src/foam/nanos/export/GoogleSheetsExportService.js]: fixed

**Pull request merged**